controllers/users_config_map_controller.go

```diff
@@ -0,0 +1,303 @@
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.ConfigMap{}, builder.WithPredicates(predicate.Funcs{
+			CreateFunc: func(e event.CreateEvent) bool {
+				new, ok := e.Object.(*corev1.ConfigMap)
+				if !ok {
+					return false
+				}
+				return IsUsersConfigMap(new)
+			},
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				new, ok := e.ObjectNew.(*corev1.ConfigMap)
+				if !ok {
+					return false
+				}
+				return IsUsersConfigMap(new)
+			},
+			DeleteFunc: func(e event.DeleteEvent) bool {
+				new, ok := e.Object.(*corev1.ConfigMap)
+				if !ok {
+					return false
+				}
+				return IsUsersConfigMap(new)
+			},
+			GenericFunc: func(e event.GenericEvent) bool {
+				new, ok := e.Object.(*corev1.ConfigMap)
+				if !ok {
+					return false
+				}
+				return IsUsersConfigMap(new)
+			},
+		})).
+		Complete(r)
+}
+
+func IsUsersConfigMap(configMap *corev1.ConfigMap) bool {
+	return configMap.Name == additionalUsersConfigMapName
+}
+
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=imagerepositories,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=core,resources=configmaps,verbs=get;list;watch;update
+
+func (r *QuayUsersConfigMapReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := ctrllog.FromContext(ctx).WithName("ConfigMap")
+	ctx = ctrllog.IntoContext(ctx, log)
+
+	// fetch the config map instance
+	configMap := &corev1.ConfigMap{}
+	if err := r.Client.Get(ctx, req.NamespacedName, configMap); err != nil {
+		if errors.IsNotFound(err) {
+			// The object is deleted, nothing to do
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "failed to get config map", l.Action, l.ActionView)
+		return ctrl.Result{}, err
+	}
+
+	if !controllerutil.ContainsFinalizer(configMap, ConfigMapFinalizer) {
+		controllerutil.AddFinalizer(configMap, ConfigMapFinalizer)
+
+		if err := r.Client.Update(ctx, configMap); err != nil {
+			log.Error(err, "failed to add finalizer to config map", "ConfigMapName", additionalUsersConfigMapName, "Finalizer", ConfigMapFinalizer, l.Action, l.ActionUpdate)
+			return ctrl.Result{}, err
+		}
+		log.Info("finalizer added to configmap")
+		// leave all other steps to reconcile from update action
+		return ctrl.Result{}, nil
+	}
+
+	teamName := getQuayTeamName(configMap.Namespace)
+	removeTeam := false
+
+	if !configMap.DeletionTimestamp.IsZero() {
+		removeTeam = true
+		log.Info("Config map with additional users was removed, will delete team", "TeamName", teamName)
+	}
+
+	var additionalUsers []string
+	if !removeTeam {
+		// get additional users from config map
+		additionalUsersStr, usersExist := configMap.Data[additionalUsersConfigMapKey]
+		if !usersExist {
+			log.Info("Config map with additional users doesn't have the key", "ConfigMapName", additionalUsersConfigMapName, "ConfigMapKey", additionalUsersConfigMapKey, l.Action, l.ActionView)
+			removeTeam = true
```

Comment:
The team will not be recreated if the key in config map is removed and added back.
Do we need to bind the team to the field or whole config map would be ok too, WDYT?

Reply:
mmm??  

if config map is updated => key removed it will trigger reconcile which removes team

if config map is updated => key added back, it will trigger another reconcile which creates team

Reply:
Yes, correct, I overlooked.